src/App.tsx

```diff
@@ -1,61 +1,102 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './components/types/Todo';
+
+export const App: React.FC = () => {
+  const [users] = useState(usersFromServer);
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTitleError(!title);
+    setUserIdError(!selectedUserId);
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id), 0);
```

Comment:
The use of `Math.max(...todos.map(todo => todo.id), 0)` is correct in terms of syntax, but it could lead to incorrect ID assignment if the `todos` array is empty. In such a case, `Math.max` will return `-Infinity`, and adding 1 will result in `NaN`. Consider initializing `maxId` with `0` if the `todos` array is empty.